### PR TITLE
Add DisplayFaction details to the replay metadata.

### LINKED ISSUE #18872: Save selected faction info in replay metadata
Currently only the resolved Faction state is stored, and whether it was random or not. There is no possibility to differentiate between "Random", "Random Allies" and "Random Soviet" picks,, they all are stored as Random.

Here's an example of Utility output:
```
c:\Games\OpenRA (dev)>OpenRA.Utility.exe ra --replay-metadata ra-2020-11-29T105016Z.orarep
ra-2020-11-29T105016Z.orarep:
        Mod: ra
        Version: devtest-20201128
        MapUid: 0df90cac574f501a38bd25cae664e173d56aaa10
        MapTitle: Shadowfiend II RAGL S9
        FinalGameTick: 39411
        StartTimeUtc: 2020-11-29 13-50-17
        EndTimeUtc: 2020-11-29 14-17-53
        DisabledSpawnPoints:
        Players:
                0:
                        ClientIndex: 175
                        Name: netnazgul
                        IsHuman: True
                        IsBot: False
                        FactionName: Russia
                        FactionId: russia
                        Color: 5D557A
                        Team: 0
                        SpawnPoint: 2
                        IsRandomFaction: True
                        IsRandomSpawnPoint: True
                        Fingerprint: 791cc59c91395d1b73b77d0470c4a90ad3ecf0ae
                        Outcome: Lost
                        OutcomeTimestampUtc: 2020-11-29 14-16-33
                        DisconnectFrame: 0
                1:
                        ClientIndex: 200
                        Name: U
                        IsHuman: True
                        IsBot: False
                        FactionName: Ukraine
                        FactionId: ukraine
                        Color: F3E506
                        Team: 0
                        SpawnPoint: 1
                        IsRandomFaction: True
                        IsRandomSpawnPoint: True
                        Fingerprint: 5725c7a8419ba1c04a2200030f356ebb98f0d4f0
                        Outcome: Won
                        OutcomeTimestampUtc: 2020-11-29 14-16-33
                        DisconnectFrame: 0
```

And this is one of the metadata frames from that replay just before the game is started:
```
SyncInfo йDisabledSpawnPoints:
Client@175:
    Index: 175
    PreferredColor: 5D557A
    Color: 5D557A
    Faction: RandomSoviet
    SpawnPoint: 0
    Name: netnazgul
    AnonymizedIPAddress: 8.40.107.*
    Location: Poland
    State: Ready
    Team: 0
    Slot: Multi0
    Bot:
    BotControllerClientIndex: 0
    IsAdmin: True
    Fingerprint: 791cc59c91395d1b73b77d0470c4a90ad3ecf0ae
```

Another field (i.e. `FactionSelected`) might be added to metadata footer.

### PR DESCRIPTION
**Pull request by pchote**
Fixes #18872.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/World/CreateMPPlayers.cs [hunk 1]
```diff
@@ -51,6 +51,7 @@ void ICreatePlayersInfo.CreateServerPlayers(MapPreview map, Session lobbyInfo, L
 
 				var clientFaction = factions.First(f => client.Faction == f.InternalName);
 				var resolvedFaction = Player.ResolveFaction(client.Faction, factions, playerRandom, !kv.Value.LockFaction);
+				var resolvedDisplayFaction = factions.FirstOrDefault(f => f.InternalName == client.Faction) ?? factions.First();
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/World/CreateMPPlayers.cs]: This appears to be the exact same as `clientFaction`?

File: OpenRA.Game/GameInformation.cs [hunk 2]
```diff
@@ -157,6 +159,10 @@ public class Player
 			public string FactionId;
 			public Color Color;
 
+			/// <summary>The faction (including Random, etc) that was selected in the lobby.</summary>
```
Comment (abcdefg30) [inline OpenRA.Game/GameInformation.cs]: Actually, while I'm already requesting changes, `etc` is missing the full-stop.

**Comment by pchote** [comment]
Fixed.

**Review by abcdefg30 — APPROVED** [review]
Code changes lgtm and this is already used in the devtest versions.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/c82bd924ba61611f137c64cd03abfe586c4f1354...41bf482b380615c30ffcfb65ccaf6291a0156966

**Pull request merged**